compiler/noirc_evaluator/src/ssa/plonky2_gen/asm_writer.rs

```diff
@@ -576,6 +616,42 @@ impl AsmWriter {
         self.comment(format!("lessthan end (result = {})", BoolTargetDisplay { t: result }));
     }
 
+    fn add_debug_trace_source_file_line(&mut self, name: String, line_number: usize) {
+        let dtlist = self.debug_trace_list.as_mut().unwrap();
+
+        if let Some(prev_dsp) = &self.prev_dsp {
+            if let Some(last_range_vec) = dtlist.source_map.get_mut(prev_dsp) {
+                last_range_vec.last_mut().unwrap().end = Some(dtlist.list.len() - 1);
+            }
+        }
```

Comment:
Feels like we could do better error handling here too, but I'm out of juice to make code suggestions at this point.

Reply:
Added panic on else in the inner if, because it indicates an internal compiler error: https://github.com/blocksense-network/noir/pull/96/commits/c6a129a38f98aee14d73e184afb6c9f6f1c3507d

Reply:
Done: https://github.com/blocksense-network/noir/pull/96/commits/e2fd9c6ce839c8a22e1083a639189c8ee41bfbe4